src/Storages/AlterCommands.cpp

```diff
@@ -1090,7 +1090,14 @@ void AlterCommands::validate(const StoragePtr & table, ContextPtr context) const
                                                              "in a single ALTER query", backQuote(column_name));
 
             if (command.codec)
```

Comment:
It seems that the code above this in line 1076 is already doing a loop over all_columns. Would it more efficient if this new loop is merged with above? Something like is_alias is set as well in the loop at line 1076.

Reply:
thanks, removed loop from the logic